src/pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -105,6 +105,9 @@ def _set_ckpt_path(
         ft_checkpoints = [cb for cb in self.trainer.callbacks if isinstance(cb, _FaultToleranceCheckpoint)]
         fn = state_fn.value
 
+        if ckpt_path is None and SLURMEnvironment.detect() and self._hpc_resume_path is not None:
```

Comment:
Do you think it's needed to check for `SLURMEnvironment.detect()`?

Reply:
I think it's safer in case somebody did a rsync of a SLURM filesystem, but this is an extreme edge case

Reply:
I agree with @carmocca, as far as I understand HPC checkpoints are useless outside of SLURM environment